frontend/src/context/MainContext.js

```diff
@@ -0,0 +1,22 @@
+import { createContext, useState } from "react";
+
+const MainContext = createContext();
+
+export const MainProvider = ({ children }) => {
+  //This global state is to track if the app is in "edit" mode;
+
+  const [search, setSearch] = useState("Search by name");
```

Comment:
It would be better to set the default state to be `""` here so that when user want to search for something they don't have to delete the entire `"Search by name"` phrase before entering the label they want to find.